presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/InternalResourceGroup.java

```diff
@@ -131,17 +137,43 @@ protected InternalResourceGroup(Optional<InternalResourceGroup> parent, String n
     public ResourceGroupInfo getInfo()
     {
         synchronized (root) {
+            checkState(!subGroups.isEmpty() || (descendantRunningQueries == 0 && descendantQueuedQueries == 0), "Leaf resource group has descendant queries.");
```

Comment:
Seems like descendant is the right word. https://www.merriam-webster.com/dictionary/descendant